### PR TITLE
upgrade upload

### PR DESCRIPTION
**Pull request by baiyaaaaa**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.
1. 支持手动上传
2. 增加 `listType` API
3. 增加 `onChange` 钩子
4. 部分 API 更新

### CODE REVIEW

File: examples/docs/zh-CN/upload.md [hunk 1]
```diff
@@ -165,17 +295,19 @@
...
-| on-error | 可选参数, 文件上传失败时的钩子 | function(err, response, file) | — | — |
+| on-error | 可选参数, 文件上传失败时的钩子 | function(err, file, fileList) | — | — |
 | on-progress | 可选参数, 文件上传时的钩子 | function(event, file, fileList) | — | — |
+| on-change | 可选参数, 文件状态改变时的钩子 | function(file, fileList) | — | — |
 | before-upload | 可选参数, 上传文件之前的钩子，参数为上传的文件，若返回 false 或者 Promise 则停止上传。 | function(file) | — | — |
-| thumbnail-mode | 是否设置为图片模式，该模式下会显示图片缩略图 | boolean | — | false |
-| default-file-list | 默认已上传的文件列表, 例如: [{name: 'food.jpeg', url: 'https://fuss10.elemecdn.com/3/63/4e7f3a15429bfda99bce42a18cdd1jpeg.jpeg?imageMogr2/thumbnail/360x360/format/webp/quality/100'}] | array | — | [] |
+| list-type | 文件列表的类型 | string | text/picture/picture-card | text |
+| auto-upload | 是否在选取文件后立即进行上传 | boolean | — | true |
+| fileList | 上传的文件列表, 例如: [{name: 'food.jpeg', url: 'https://fuss10.elemecdn.com/3/63/4e7f3a15429bfda99bce42a18cdd1jpeg.jpeg?imageMogr2/thumbnail/360x360/format/webp/quality/100'}] | array | — | [] |
```
Comment (QingWei-Li) [inline examples/docs/zh-CN/upload.md]: file-list

File: packages/upload/src/index.vue [hunk 2]
```diff
@@ -34,9 +33,11 @@ export default {
       type: String,
       default: 'file'
     },
+    drag: Boolean,
+    dragger: Boolean,
     withCredentials: Boolean,
     thumbnailMode: Boolean,
-    showUploadList: {
```
Comment (QingWei-Li) [inline packages/upload/src/index.vue]: 老 API `showUploadList` 应该同样兼容或者用 migrating.js 给提示信息

File: packages/upload/src/index.vue [hunk 3]
```diff
@@ -70,127 +70,137 @@ export default {
       type: Function,
       default: noop
     },
-    defaultFileList: {
+    fileList: {
```
Comment (QingWei-Li) [inline packages/upload/src/index.vue]: 同上

File: packages/upload/src/upload-dragger.vue [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div
+    class="el-upload-dragger"
+    :class="{
+      'is-dragOver': dragOver
```
Comment (QingWei-Li) [inline packages/upload/src/upload-dragger.vue]: 额 貌似 css 命名一直都是小写的

**Comment by baiyaaaaa** [comment]
done

File: packages/upload/src/index.vue [hunk 5]
```diff
@@ -70,127 +73,148 @@ export default {
...
+      this.uploadFiles
+        .filter(file => file.status === 'ready')
+        .forEach(file => {
+          this.$refs['upload-inner'].upload(file.raw, file);
+        });
+    },
+    getMigratingConfig() {
+      return {
+        props: {
+          'default-file-list': 'default-file-list is rename to file-list.',
```
Comment (Leopoldthecoder) [inline packages/upload/src/index.vue]: renamed

File: packages/upload/src/index.vue [hunk 6]
```diff
@@ -70,127 +73,148 @@ export default {
...
+          'show-upload-list': 'show-file-list is rename to show-file-list.'
```
Comment (Leopoldthecoder) [inline packages/upload/src/index.vue]: renamed

**Pull request merged**